langchain/cache.py

```diff
@@ -336,6 +336,7 @@ def _new_gptcache(self, llm_string: str) -> Any:
                 self.init_gptcache_func(_gptcache, llm_string)  # type: ignore[call-arg]
             else:
                 self.init_gptcache_func(_gptcache)  # type: ignore[call-arg]
+            self.gptcache_dict[llm_string] = _gptcache
```

Comment:
i think it should be put in the line 345

Reply:
please check my 2nd commit, is it correct?

Reply:
It's nice, thanks your fix